File: parsers/src/test/java/wcl/gpromanager/parsers/testutils/HtmlFileLoader.java
package wcl.gpromanager.parsers.testutils;

import org.apache.commons.io.Charsets;
import org.apache.commons.io.IOUtils;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import static org.junit.jupiter.api.Assertions.fail;

public class HtmlFileLoader {
    private HtmlFileLoader() {}

    public static String loadHtmlFile(final String htmlFileName)  {
        String fileContent = "";
        try (var inputStream = HtmlFileLoader.class.getResourceAsStream(htmlFileName)) {
            fileContent = IOUtils.toString(inputStream, "UTF-8");
        } catch (IOException e) {
            fail("Error loading file " + htmlFileName);
        }
        return fileContent;

    }
}

File: parsers/src/test/kotlin/wcl/gpromanager/parsers/testutils/HtmlFileLoader.kt
package wcl.gpromanager.parsers.testutils

import org.apache.commons.io.IOUtils
import org.junit.jupiter.api.Assertions.*
import java.io.IOException

object HtmlFileLoader {
    @JvmStatic
    fun loadHtmlFile(htmlFileName: String): String {
        var fileContent = ""
        try {
            HtmlFileLoader::class.java.getResourceAsStream(htmlFileName).use { inputStream ->
                fileContent = IOUtils.toString(inputStream, "UTF-8")
            }
        } catch (e: IOException) {
            fail<Any>("Error loading file $htmlFileName")
        }
        return fileContent
    }
}
